src/types.jl

```diff
@@ -1,143 +1,219 @@
+            acceleration = CPU1(),
+            embedding_dims = Dict{Symbol, Real}(),
+        ) where {B, O, L}
+
+            model = $Model{B, O, L}(
+                builder,
+                optimiser,
+                loss,
+                epochs,
+                batch_size,
+                lambda,
+                alpha,
+                rng,
+                optimiser_changes_trigger_retraining,
+                acceleration,
+                embedding_dims,
+            )
+
+            message = clean!(model)
+            isempty(message) || @warn message
+
+            return model
+        end
+
+    end |> eval
 
 end
 
 const Regressor =
-  Union{NeuralNetworkRegressor,MultitargetNeuralNetworkRegressor}
+    Union{NeuralNetworkRegressor, MultitargetNeuralNetworkRegressor}
+
+
+# Separately define the ImageClassifier
+mutable struct ImageClassifier{B, F, O, L} <: MLJFluxProbabilistic
+    builder::B
+    finaliser::F
+    optimiser::O
+    loss::L
+    epochs::Int
+    batch_size::Int
+    lambda::Float64
+    alpha::Float64
+    rng::Union{AbstractRNG, Int64}
+    optimiser_changes_trigger_retraining::Bool
+    acceleration::AbstractResource
+end
+
+function ImageClassifier(
+    ; builder::B = image_builder(VGGHack),
+    finaliser::F = Flux.softmax,
+    optimiser::O = Optimisers.Adam(),
+    loss::L = Flux.crossentropy,
+    epochs = 10,
+    batch_size = 1,
+    lambda = 0,
+    alpha = 0,
+    rng = Random.default_rng(),
+    optimiser_changes_trigger_retraining = false,
+    acceleration = CPU1(),
+) where {B, F, O, L}
+    model = ImageClassifier{B, F, O, L}(
+        builder,
+        finaliser,
+        optimiser,
+        loss,
+        epochs,
+        batch_size,
+        lambda,
+        alpha,
+        rng,
+        optimiser_changes_trigger_retraining,
+        acceleration,
+    )
+
+    message = clean!(model)
+    isempty(message) || @warn message
+
+    return model
+end
+
 
 MMI.metadata_pkg.(
-  (
-      NeuralNetworkRegressor,
-      MultitargetNeuralNetworkRegressor,
-      NeuralNetworkClassifier,
-      ImageClassifier,
-      NeuralNetworkBinaryClassifier,
-  ),
-    name="MLJFlux",
-    uuid="094fc8d1-fd35-5302-93ea-dabda2abf845",
-    url="https://github.com/alan-turing-institute/MLJFlux.jl",
-    julia=true,
-    license="MIT",
+    (
+        NeuralNetworkRegressor,
+        MultitargetNeuralNetworkRegressor,
+        NeuralNetworkClassifier,
+        ImageClassifier,
+        NeuralNetworkBinaryClassifier,
+    ),
+    name = "MLJFlux",
+    uuid = "094fc8d1-fd35-5302-93ea-dabda2abf845",
+    url = "https://github.com/alan-turing-institute/MLJFlux.jl",
+    julia = true,
+    license = "MIT",
 )
 
+const MODELSUPPORTDOC = """
+This model supports categorical columns in the input table. If present, such columns are embedded into
+dense vectors by the use of an additional `EntityEmbedder` layer after the input 
+as described in Entity Embeddings of Categorical Variables by Cheng Guo, Felix Berkhahn.
```

Comment:
This reference needs a year, journal name and volume number.

Reply:
@ablaom it doesn't seem to be published in a journal/conference for some reason. I will do `as described in Entity Embeddings of Categorical Variables by Cheng Guo, Felix Berkhahn, arXiv, 2016.`.

Reply:
Done in f7abf07287aa0478ca72444d3864910eb564c19e